service/collector/app_resource.go

```diff
@@ -38,19 +61,63 @@ func NewAppResource(config AppResourceConfig) (*AppResource, error) {
 		g8sClient: config.G8sClient,
 		k8sClient: config.K8sClient,
 		logger:    config.Logger,
+
+		watchNamespace: config.WatchNamespace,
 	}
 
 	return c, nil
 }
 
+func (c *AppResource) collectAppStatus(ctx context.Context, ch chan<- prometheus.Metric) {
+	r, err := c.g8sClient.ApplicationV1alpha1().Apps(c.watchNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		c.logger.LogCtx(ctx, "level", "error", "message", "could not get apps", "stack", fmt.Sprintf("%#v", err))
+	}
+
+	for _, app := range r.Items {
+		ch <- prometheus.MustNewConstMetric(
+			appDesc,
+			prometheus.GaugeValue,
+			gaugeValue,
+			app.Name,
+			app.Namespace,
+			app.Status.Release.Status,
+			app.Status.Version,
+		)
+
+	}
+
+}
+
 // Collect is the main metrics collection function.
 func (c *AppResource) Collect(ch chan<- prometheus.Metric) error {
-	// TODO
+	ctx := context.Background()
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting metrics")
+
+	collectFuncs := []func(context.Context, chan<- prometheus.Metric){
+		c.collectAppStatus,
+	}
+
+	var wg sync.WaitGroup
+
+	for _, collectFunc := range collectFuncs {
+		wg.Add(1)
+
+		go func(collectFunc func(ctx context.Context, ch chan<- prometheus.Metric)) {
+			defer wg.Done()
+			collectFunc(ctx, ch)
+		}(collectFunc)
```

Comment:
All this `collectFunc` stuff is not necessary. We do this in the `aws-operator` because of the different AWS accounts we need to cover for BYOC. This is ugly in general and not necessary for apps. The logic of `collectAppStatus` should be what we want.

Reply:
Thanks, I will delete the whole waitgroup section of this area.